cmd/tempo-serverless/handler.go

```diff
@@ -0,0 +1,208 @@
+	delete(searchReq.Tags, api.URLParamIndexPageSize)
+	delete(searchReq.Tags, api.URLParamTotalRecords)
+	delete(searchReq.Tags, api.URLParamTenant)
+	delete(searchReq.Tags, api.URLParamDataEncoding)
+	delete(searchReq.Tags, api.URLParamVersion)
+
+	start, end, limit, err := api.ParseBackendSearch(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	startPage, totalPages, blockID, err := api.ParseBackendSearchQuerier(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	enc, dataEncoding, indexPageSize, totalRecords, tenant, version, err := api.ParseBackendSearchServerless(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// load local backend config file. by convention this must be in the same folder at ./config.json
+	reader, err := loadBackend()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// fake out meta, we're only filling in the fields here we need
+	// which is kind of cheating
+	meta := &backend.BlockMeta{
+		Version:       version,
+		TenantID:      tenant,
+		Encoding:      enc,
+		IndexPageSize: indexPageSize,
+		TotalRecords:  totalRecords,
+		BlockID:       blockID,
+		DataEncoding:  dataEncoding,
+	}
+
+	block, err := encoding.NewBackendBlock(meta, reader)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// todo(search): pull 1MB hardcoded param and 10000 prefetch from somewhere
+	//               the below iterator setup and loop exists in the querier, serverless and tempo-cli. see if there is an opportunity to consolidate
+	iter, err := block.PartialIterator(1_000_000, int(startPage), int(totalPages))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	iter = encoding.NewPrefetchIterator(r.Context(), iter, 10000)
+
+	resp := &tempopb.SearchResponse{
+		Metrics: &tempopb.SearchMetrics{},
+	}
+	for {
+		id, obj, err := iter.Next(r.Context())
+		if err == io.EOF || id == nil || obj == nil {
+			break
+		}
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		resp.Metrics.InspectedTraces++
+		resp.Metrics.InspectedBytes += uint64(len(obj))
+
+		metadata, err := model.Matches(id, obj, dataEncoding, uint32(start), uint32(end), searchReq)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		if metadata == nil {
+			continue
+		}
+
+		resp.Traces = append(resp.Traces, metadata)
+		if len(resp.Traces) >= int(limit) {
+			break
+		}
+	}
+
+	marshaller := &jsonpb.Marshaler{}
+	err = marshaller.Marshal(w, resp)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func loadBackend() (backend.Reader, error) {
+	// horrible viper dance since it won't unmarshal to a struct from env: https://github.com/spf13/viper/issues/188
```

Comment:
I can't find a way in Google Cloud Functions to deliver a config file alongside the binary/code. It might be impossible. However, setting environment variables is supported across all serverless cloud functions which is why we are supporting that.